annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/AnnotationController.java

```diff
@@ -111,35 +103,56 @@ public AnnotationController(VariantAnnotationService variantAnnotationService,
-
-            postEnrichmentService.registerEnricher(isoformOverrideSource, enricher);
-        }
+    @ApiOperation(value = "Retrieves VEP annotation for the provided list of variants",
+        hidden = true,
+        nickname = "postVariantAnnotation")
+    @RequestMapping(value = "/hgvs",
+        method = RequestMethod.POST,
+        produces = "application/json")
+    @Deprecated
+    public List<VariantAnnotation> postVariantAnnotation(
+        @RequestParam
+        @ApiParam(value="Comma separated list of variants. For example X:g.66937331T>A,17:g.41242962->GA",
+            required = true,
+            allowMultiple = true)
+            List<String> variants,
+        @RequestParam(required = false)
+        @ApiParam(value="Isoform override source. For example uniprot",
+            required = false)
+            String isoformOverrideSource,
+        @RequestParam(required = false)
+        @ApiParam(value="Comma separated list of fields to include (case-sensitive!). " +
+            "For example: hotspots,mutation_assessor", required = false, defaultValue = "hotspots,mutation_assessor")
+            List<String> fields)
+    {
+        return fetchVariantAnnotationPOST(variants, isoformOverrideSource, fields);
+    }
 
-        if (fields != null && fields.contains("hotspots"))
-        {
-            AnnotationEnricher enricher = new HotspotAnnotationEnricher(hotspotService, true);
-            postEnrichmentService.registerEnricher("cancerHotspots", enricher);
-        }
-        if (fields != null && fields.contains("mutation_assessor"))
-        {
-            AnnotationEnricher enricher = new MutationAssessorAnnotationEnricher(mutationAssessorService);
-            postEnrichmentService.registerEnricher("mutation_assessor", enricher);
-        }
+    @ApiOperation(value = "Retrieves VEP annotation for the provided list of variants",
+        nickname = "fetchVariantAnnotationPOST")
+    @RequestMapping(value = "/annotation",
+        method = RequestMethod.POST,
+        produces = "application/json")
+    public List<VariantAnnotation> fetchVariantAnnotationPOST(
+        @RequestBody
+        @ApiParam(value="List of variants. For example [\"X:g.66937331T>A\",\"17:g.41242962->GA\"]",
+            required = true)
+            List<String> variants,
+        @RequestParam(required = false)
+        @ApiParam(value="Isoform override source. For example uniprot",
+            required = false)
+            String isoformOverrideSource,
+        @RequestParam(required = false)
+        @ApiParam(value="Comma separated list of fields to include (case-sensitive!). " +
+            "For example: hotspots,mutation_assessor", required = false, defaultValue = "hotspots,mutation_assessor")
+            List<String> fields)
+    {
+		List<VariantAnnotation> variantAnnotations = new ArrayList<>();
+        EnrichmentService postEnrichmentService = this.initPostEnrichmentService(isoformOverrideSource, fields);
```

Comment:
I don't fully understand this pattern - if *fields* comes in with every request and is important for every enrichAnnotation() call, then why not pass fields in as an argument to the service instead of instantiating a new instance of the service with the fields passed in as a constructor argument? This seems like a waste of heap to me; but I'm sure there is  reason (maybe some inheritance / polymorphism issue). We could still hold the service instances in a cache, so we don't need to keep instantiating them perhaps.

Reply:
Oh, this one is a bit tricky. Actually, how we enrich the core VEP annotaiton depends on the `fields`. With this approach we re-instantiate the service and register enrichers for each request. It is like re-configuring the enrichment service for each query separately. But, I think we can optimize this way we configure the enrichment per query. It may not be necessary to re-instantiate the service each time.